_posts/en/newsletters/2020-09-16-newsletter.md

```diff
@@ -0,0 +1,216 @@
+  of memory and crash.
+
+    Several of the affected programs had code that limited the number of
+    inventories that should've been queued, but the researchers were
+    able to circumvent those protections.  As a bug that could only crash
+    nodes, it could not be used to directly steal from affected nodes.
+    However, it could be used to execute [eclipse attacks][topic eclipse
+    attacks] that could ultimately steal money.  However, there's no
+    indication the attack was ever used in public.
+
+    All users are encouraged to upgrade to the latest released version
+    of their preferred full node software, or to review the list of
+    vulnerable and patched versions on the [website][invdos site].
+
+- **Stealing onchain fees from LN HTLCs:** Antoine Riard [posted][riard
+  post] to the Lightning-Dev mailing list a potential vulnerability in a
+  recent update to the LN specification (see [Newsletter #112][news112
+  bolts688]). When LN payments are made according to the updated
+  specification, each remote party includes a flag in the signature for
+  their half of the 2-of-2 multisig script.  The flag, called
+  `SIGHASH_SINGLE|SIGHASH_ANYONECANPAY`, allows the payment (called an
+  HTLC) to be placed into a transaction containing other inputs and
+  outputs.  This is designed to allow multiple HTLCs  to be bundled
+  together into a transaction that also contains an extra input that
+  pays any additional transaction fees and an optional extra output that
+  collects any money left over from paying those extra fees.  This makes
+  it possible to wait until those HTLCs need to be confirmed before
+  choosing what feerate to pay.
+
+        Inputs             Fee         Outputs
+        (Implicit values)  (Implicit)
+
+        HTLC(0.1 BTC) -----0.01 BTC--> 0.09 BTC to Alice (payment)
+        HTLC(0.1 BTC) -----0.01 BTC--> 0.09 BTC to Alice (payment)
+        Alice(0.1 BTC) ----0.05 BTC--> 0.05 BTC to Alice (change)
+
+        FIXME:image-ify
+
+    However, Riard notes that the specification's earlier mechanism of
+    committing to variable feerates for HTLCs is still available.  This
+    can allow an attacker to commit to a high fee for multiple HTLCs
+    and then construct a transaction that combines those HTLCs with just
+    an additional output that claims those fees.  Riard's email
+    describes how to use this basic technique in a more complex attack
+    to maximize the amount stolen.
+
+        Inputs             Fee         Outputs
+        (Implicit values)  (Implicit)
+
+        HTLC(0.1 BTC) -----0.09 BTC--> 0.01 BTC to Alice (payment)
+        HTLC(0.1 BTC) -----0.09 BTC--> 0.01 BTC to Alice (payment)
+                                       0.18 BTC to Mallory (theft)
+
+        FIXME:image-ify
+
+    Several solutions were proposed, the simplest of which might be to
+    have HTLCs only pay a minimal relay fee---requiring the party
+    broadcasting the HTLCs add any additional necessary fees.  No LN
```

Comment:
maybe not strictly necessary, but might be more readable to replace the subjonctive with the infinitive: /add/to add/